### PR TITLE
Correct instruction caching on cAVS 2.5 non-primary cores

### PR DESCRIPTION
**Pull request by andyross**
The MP startup path on cAVS 2.5 was setting up the data caching behavior for the Xtensa region protection option, but forgetting the instruction cache.  This is clearly incorrect, and per the SOF team visible in some performance regressions that have been visible in Zephyr builds.

Pull request also includes a bit of refactoring to enable this (code coming right behind this finishes a big rework of the bootloader layer which unifies CPU initialization and code entry to a single path across all the platforms and for all cores).  The last patch contains the meat of the issue.

### CODE REVIEW

**Comment by andyross** [comment]
Really this is part of the bootloader/IMR/platform-unification work, but I stumbled on the icache regression along the way, so it makes sense to push this out early so SOF can validate.

**Review by dcpleung — APPROVED** [review]

**Comment by andyross** [comment]
Oh, and for those reviewing: I pushed out the remainder of the-subset-of-my-bootloader-tree-relevant-to-upstream (of which this PR is the bottom four patches) at https://github.com/andyross/zephyr/tree/cavs-boot-tmp

I'm going to be submitting that very soon too, though it still needs a bunch of splitting/squashing work.  And there's a final batch coming behind that which completes the "move bootloader into a single Zephyr link".  But that last bit isn't working generically yet, only with an internal frankentree of mine. :)

**Comment by andyross** [comment]
Hrm.  Build errors confirmed.  There's some stuff that's building in west but not twister.  Fun fun, I just love glitches like that...

File: soc/xtensa/intel_adsp/common/include/cpu_init.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2021 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __INTEL_ADSP_CPU_INIT_H
+#define __INTEL_ADSP_CPU_INIT_H
+
+#include <xtensa/config/core-isa.h>
+
+#define CxL1CCAP (*(volatile uint32_t *)0x9F080080)
```
Comment (teburd) [inline soc/xtensa/intel_adsp/common/include/cpu_init.h:9]: I wish we could link to the ISA docs for stuff like this, but alas...

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Tested booth on one cavs2.5 platform with SOF app and at least boots fine. A small perf impact can also be seen https://github.com/thesofproject/sof/issues/5050#issuecomment-992738184

**Comment by andyross** [comment]
Heh.  So it was a real bug and the fix improves performance, but not by very much.  Figures. :(  Yeah, I guess it was always a little dubious that streaming DSP code would be limited by instruction fetch bandwidth.

**Comment by andyross** [comment]
OK, unfortunately I've made a little mess.  I fixed the build errors that are reported in this PR (just a linker warning that snuck past me, basically) . But I've done it in the superset PR #41098 instead of here, and at a later stage in the series.  It's sort of a mess to unsquash.

Given that it looks like that might merge first anyway, I'm going to ignore the problem for now.  If needed I can come back and do a targetted fix here and then rebase the other PR, but... maybe I won't have to.

**Comment by andyross** [comment]
Close this.  The superset PR merged instead.

**Pull request closed without merging**